homeassistant/components/mcp_server/http.py

```diff
@@ -168,3 +218,64 @@ async def post(
         _LOGGER.debug("Received client message: %s", message)
         await session.read_stream_writer.send(SessionMessage(message))
         return web.Response(status=200)
+
+
+class ModelContextProtocolStreamableView(HomeAssistantView):
+    """Model Context Protocol Streamable HTTP endpoint."""
+
+    name = f"{DOMAIN}:streamable"
+    url = STREAMABLE_API
+
+    async def get(self, request: web.Request) -> web.StreamResponse:
+        """Handle unsupported methods."""
+        return web.Response(
+            status=HTTPStatus.METHOD_NOT_ALLOWED, text="Only POST method is supported"
+        )
+
+    async def post(self, request: web.Request) -> web.StreamResponse:
+        """Process JSON-RPC messages for the Model Context Protocol."""
+        hass = request.app[KEY_HASS]
+        entry = async_get_config_entry(hass)
+
+        # The request must include a JSON-RPC message
+        if CONTENT_TYPE_JSON not in request.headers.get("accept", ""):
+            raise HTTPBadRequest(text=f"Client must accept {CONTENT_TYPE_JSON}")
+        if request.content_type != CONTENT_TYPE_JSON:
+            raise HTTPBadRequest(text=f"Content-Type must be {CONTENT_TYPE_JSON}")
+        json_data = await request.json()
+        try:
```

Comment:
`request.json()` can raise if it's not valid JSON
```suggestion
        try:
            json_data = await request.json()
```